tests/test_ext_commands.py

```diff
@@ -134,6 +137,31 @@ def test_sanitize_repo_commits_non_processed_files_to_target_branch(
     )
 
 
+def test_sanitize_repo_returns_fail_when_repo_has_staged_changes(
+    sanitizer_config, fake_repo
+):
+    tracked_file = fake_repo.file_infos[0].abspath
+    tracked_file.write_text("this is the new text!")
+    fake_repo.repo.git.add(tracked_file)
+
+    result = execute_sanitize_repo(
+        f"--file-list {fake_repo.file_list_path} "
+        f"--repo-root {fake_repo.path} "
+        "--no-commit"
+    )
+
+    assert result.status == plug.Status.ERROR
+
```

Comment:
This test is very nice, but I think we should also add an assert on the message, to make sure we actually get the expected error.
```suggestion
assert "uncommitted staged files" in result.msg

```

PS: This might also be a good time to fix the typo in the implementation which reads `uncommited` (a single t, there should be two) :)